runtime/calamari/Cargo.toml

```diff
@@ -116,12 +117,14 @@ runtime-benchmarks = [
   'frame-system-benchmarking',
   'frame-support/runtime-benchmarks',
   'frame-system/runtime-benchmarks',
+  'manta-collator-selection/runtime-benchmarks',
+  'nimbus-primitives/runtime-benchmarks',
+  'pallet-author-inherent/runtime-benchmarks',
```

Comment:
You need   'cumulus-pallet-parachain-system/runtime-benchmarks' for the benchmark to work.
[This impl](https://github.com/paritytech/cumulus/blob/3bb82e5126362f9cf3952c60b737c3b71b0a136b/pallets/parachain-system/src/lib.rs#L1113-L1126) of `BlockNumberProvider` is never used since the feature is not enabled.